src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -6827,28 +6831,28 @@ void Boss07_Remains_IntroCutscene(Boss07* this, PlayState* play) {
                 this->targetPos.x = sRemainsEndTarget[MAJORA_GET_TYPE(&this->actor) - MAJORA_TYPE_REMAINS].x;
                 this->targetPos.y = 370.0f;
                 this->targetPos.z = sRemainsEndTarget[MAJORA_GET_TYPE(&this->actor) - MAJORA_TYPE_REMAINS].z;
-                sp38 = 20.0f;
-                sp34 = 0.5f;
-                sp40 = 0x1000;
-                sp3C = 0x190;
-                phi_f2 = 0.0f;
+                speedTarget = 20.0f;
+                speedMaxStep = 0.5f;
+                speedToTargetTarget = 0x1000;
+                speedToTargetMaxStep = 0x190;
+                yOffset = 0.0f;
             } else {
-                phi_f2 = Math_SinS(this->miscTimer * 0x300) * 20.0f;
-                sp38 = 1.0f;
-                sp34 = 0.1f;
-                sp40 = 0x5DC;
-                sp3C = 0x64;
-            }
-
-            sp54 = this->targetPos.x - this->actor.world.pos.x;
-            sp50 = this->targetPos.y - this->actor.world.pos.y + phi_f2;
-            sp4C = this->targetPos.z - this->actor.world.pos.z;
-            sp30 = Math_Atan2S(sp54, sp4C);
-            sp32 = Math_Atan2S(sp50, sqrtf(SQ(sp54) + SQ(sp4C)));
-            Math_ApproachS(&this->actor.world.rot.y, sp30, 5, this->speedToTarget);
-            Math_ApproachS(&this->actor.world.rot.x, sp32, 5, this->speedToTarget);
-            Math_ApproachF(&this->speedToTarget, sp40, 1.0f, sp3C);
-            Math_ApproachF(&this->actor.speed, sp38, 1.0f, sp34);
+                yOffset = Math_SinS(this->miscTimer * 0x300) * 20.0f;
+                speedTarget = 1.0f;
+                speedMaxStep = 0.1f;
+                speedToTargetTarget = 0x5DC;
+                speedToTargetMaxStep = 0x64;
+            }
+
+            dx = this->targetPos.x - this->actor.world.pos.x;
+            dy = this->targetPos.y - this->actor.world.pos.y + yOffset;
+            dz = this->targetPos.z - this->actor.world.pos.z;
+            rotTargetY = Math_Atan2S(dx, dz);
+            rotTargetX = Math_Atan2S(dy, sqrtf(SQ(dx) + SQ(dz)));
```

Comment:
I tend to prefer `targetRotY`/`targetRotX` for these. I like how it keeps the name of the variable this is a target for ("rotX"/"rotY") together. What do you think?